### PR TITLE
QuantumEspresso

### PR DESCRIPTION
**Pull request by simonpintarelli**
QE test for eiger and todi

### CODE REVIEW

File: checks/apps/quantumespresso/quantumespresso_check.py [hunk 1]
```diff
@@ -1,136 +1,212 @@
...
+    valid_systems = ["*"]
+    build_system = "CMake"
+    sourcesdir = None
+    maintainers = ["SSA"]
+    qe_sources = fixture(qe_download, scope="session")
+    build_locally = False
+    tags = {"uenv"}
+    pwx_executable = "./mps-wrapper.sh pw.x"
+    extra_resources = {"reservation": {"reservation": "NCCL"}}
+    time_limit = "0d0h30m0s"
```
Comment (simonpintarelli) [inline checks/apps/quantumespresso/quantumespresso_check.py:67]: the `time_limit` here seems to have no effect, I had to change it in the global config for todi as workaround

File: checks/apps/quantumespresso/quantumespresso_check.py [hunk 2]
```diff
@@ -1,136 +1,223 @@
...
-    'Au-surf': {
-        'zen2': {
-            # zen2 cpu nodes: 2sockets, 8 numa, 16c/numa = 128c (no MT)
-            'energy_reference': -11427.09017218,
-            'performance_reference': [{'R': 32, 'T': 4, 'P': 102.0}]
+
+qe_references = {
+    "Au surf": {
+        "gh200": {"time_run": (14.02, None, 0.05, "s")},
+        "zen2": {"time_run": (99.45, None, 0.05, "s")}, # 1m44s
```
Comment (jgphpc) [inline checks/apps/quantumespresso/quantumespresso_check.py:18]:
```suggestion
        "zen2": {"time_run": (99.45, None, 0.05, "s")},  # 1m44s
```

File: checks/apps/quantumespresso/quantumespresso_check.py [hunk 3]
```diff
@@ -1,136 +1,223 @@
...
+# Copyright 2016-2024 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
 # ReFrame Project Developers. See the top-level LICENSE file for details.
 #
 # SPDX-License-Identifier: BSD-3-Clause
 
-import pathlib
-import re
-import sys
+import os
+import shutil
```
Comment (jgphpc) [inline checks/apps/quantumespresso/quantumespresso_check.py:8]:
```suggestion
```

**Comment by jgphpc** [comment]
please add reframe.yaml to the image, see for example recipes/cp2k/2024.3/gh200/extra/reframe.yaml

File: checks/apps/quantumespresso/src/README.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# CP2K ReFrame Tests
```
Comment (jgphpc) [inline checks/apps/quantumespresso/src/README.md:1]: cp2k or qe ?

File: checks/apps/quantumespresso/quantumespresso_check.py [hunk 5]
```diff
@@ -1,136 +1,223 @@
...
+        "zen2": {"time_run": (99.45, None, 0.05, "s")}, # 1m44s
+    },
+}
+
+
+slurm_config = {
+    "Au surf": {
+        "gh200": {
+            "nodes": 1,
+            "ntasks-per-node": 4,
```
Comment (jgphpc) [inline checks/apps/quantumespresso/quantumespresso_check.py:27]: Is mps really needed with only 4 mpi per node ? or could we add a test with more than 4 mpi ?

**Review by jgphpc — CHANGES_REQUESTED** [review]
see comments please

**Comment by simonpintarelli** [comment]
Sorry for the late response! That notification thread got lost in my emails. Will go over your comments now.

File: checks/apps/quantumespresso/quantumespresso_check.py [hunk 6]
```diff
@@ -1,136 +1,223 @@
...
+            "-DQE_ENABLE_SCALAPACK:BOOL=ON",
+            "-DQE_ENABLE_LIBXC=ON",
+            "-DQE_CLOCK_SECONDS:BOOL=OFF",
+        ]
+
+        if self.uarch == "gh200":
+            self.build_system.config_opts += [
+                "-DQE_ENABLE_CUDA=ON",
+                "-DQE_ENABLE_MPI_GPU_AWARE:BOOL=ON",
+                "-DQE_ENABLE_OPENACC=ON",
```
[the remark this answers is not in the document]
Comment (simonpintarelli) [reply checks/apps/quantumespresso/quantumespresso_check.py:118]: This has been fixed in the new version: https://github.com/eth-cscs/alps-uenv/pull/172
I'll get the PR merged.

File: checks/apps/quantumespresso/quantumespresso_check.py (hunk 5 above)
Comment (simonpintarelli) [reply checks/apps/quantumespresso/quantumespresso_check.py:27]: The wrapper is indeed not needed. QE seems to perform best with only 1 rank per gpu. 
Shall I replace it with `SLURM_GPUS_PER_TASK=1`?

**Comment by simonpintarelli** [comment]
> please add reframe.yaml to the image, for example:
> 
> ```
> cat store.yaml
..
> ```

I've deploy the new images containing reframe.yaml:
```
[todi][simonpi@todi-ln001 ~]$ uenv image find quantumespresso
uenv/version:tag                        uarch date       id               size      
quantumespresso/v7.3.1:v3               gh200 2024-12-09 1e5e5c4ce6b92fa1 6.5GB    
```
and
```
[eiger][simonpi@eiger-ln004 ~]$ uenv image find quantumespresso
uenv/version:tag                        uarch date       id               size      
quantumespresso/v7.3.1:v2               zen2  2024-12-09 a3cb63be364bb1a5 865MB     
```

**Comment by lucamar** [comment]
It seems that the requested changes were addressed by @simonpintarelli, is it possible to merge the request @jgphpc?

File: checks/apps/quantumespresso/quantumespresso_check.py (hunk 5 above)
Comment (jgphpc) [reply checks/apps/quantumespresso/quantumespresso_check.py:27]: let's keep it as it is

**Review by jgphpc — APPROVED** [review]
Tested on todi with UENV=quantumespresso/v7.3.1:v3

```
[----------] start processing checks
[ RUN      ] qe_download ~todi:normal+quantumespresso_v7.3.1_v3_develop /dafadf3d @todi:normal+quantumespresso_v7.3.1_v3_develop
[ RUN      ] QeCheckAuSurfUenvExec /7f3eabe8 @todi:normal+quantumespresso_v7.3.1_v3_modules
[       OK ] (1/4) qe_download ~todi:normal+quantumespresso_v7.3.1_v3_develop /dafadf3d @todi:normal+quantumespresso_v7.3.1_v3_develop
[ RUN      ] QeBuildTest /f0882e2b @todi:normal+quantumespresso_v7.3.1_v3_develop

[       OK ] (2/4) QeCheckAuSurfUenvExec /7f3eabe8 @todi:normal+quantumespresso_v7.3.1_v3_modules
P: time_run: 13.58 s (r:14.02, l:None, u:0.05)
[       OK ] (3/4) QeBuildTest /f0882e2b @todi:normal+quantumespresso_v7.3.1_v3_develop
[ RUN      ] QeCheckAuSurfCustomExec /108098f3 @todi:normal+quantumespresso_v7.3.1_v3_develop
[       OK ] (4/4) QeCheckAuSurfCustomExec /108098f3 @todi:normal+quantumespresso_v7.3.1_v3_develop
P: time_run: 14.09 s (r:14.02, l:None, u:0.05)
[----------] all spawned checks have finished

[  PASSED  ] Ran 4/4 test case(s) from 4 check(s) (0 failure(s), 0 skipped, 0 aborted)
```

comments: 
- `qe_download` will activate with `module load ...`  but without a uenv the modules won't be loaded (that's fine because no module needed for wget)
- building qe takes time (~22 min)

**Review by ekouts — APPROVED** [review]

**Pull request merged**